pyccel/codegen/printing/cwrappercode.py

```diff
@@ -0,0 +1,356 @@
+                            name = self.get_new_name(used_names, "real_part"))
+            imag_part = Variable(dtype = NativeReal(),
+                            name = self.get_new_name(used_names, "imag_part"))
+            cast_function_local_vars += [real_part, imag_part]
+            cast_function_body = [Assign(real_part, FuncCall('__real', cast_function_arg))]
+            cast_function_body += [Assign(imag_part, FuncCall('__imag', cast_function_arg))]
+            cast_function_body += [Assign(cast_function_result[0], FuncCall('PyComplex_FromDoubles', [real_part, imag_part]))]
+
+        cast_function_body += [Return(cast_function_result)]
+        cast_function = FunctionDef(name       = cast_function_name,
+                                    arguments  = cast_function_arg,
+                                    body       = cast_function_body,
+                                    results    = cast_function_result,
+                                    local_vars = cast_function_local_vars)
+
+        self._cast_functions_dict[cast_type] = cast_function
+
+        return cast_function
+
+    def get_PyArgParseType(self, used_names, variable):
+
+        if variable.dtype is NativeBool():
+            collect_type = NativeInteger()
+            collect_var = Variable(dtype=collect_type, precision=4,
+                name = self.get_new_name(used_names, variable.name+"_tmp"))
+            cast_function = self.get_cast_function(used_names, 'pyint_to_bool', collect_var, variable)
+            return collect_var , cast_function
+
+        if variable.dtype is NativeComplex():
+            collect_type = PyccelPyObject()
+            collect_var = Variable(dtype=collect_type, is_pointer=True,
+                name = self.get_new_name(used_names, variable.name+"_tmp"))
+            cast_function = self.get_cast_function(used_names, 'pycomplex_to_complex', collect_var, variable)
+            return collect_var , cast_function
+
+        return variable, None
+
+    def get_PyBuildValue(self, used_names, variable):
+
+        if variable.dtype is NativeBool():
+            collect_type = PyccelPyObject()
+            collect_var = Variable(dtype=collect_type, is_pointer=True,
+                name = self.get_new_name(used_names, variable.name+"_tmp"))
+            cast_function = self.get_cast_function(used_names, 'bool_to_pyobj', variable, collect_var)
+            return VariableAddress(collect_var), cast_function
+
+        if variable.dtype is NativeComplex():
+            collect_type = PyccelPyObject()
+            collect_var = Variable(dtype=collect_type, is_pointer=True,
+                name = self.get_new_name(used_names, variable.name+"_tmp"))
+            cast_function = self.get_cast_function(used_names, 'complex_to_pycomplex', variable, collect_var)
+            return VariableAddress(collect_var), cast_function
+
+        return variable, None
+
+    def _print_PyccelPyObject(self, expr):
+        return 'pyobject'
```

Comment:
Is this function used?

Reply:
Yes it is used here: https://github.com/pyccel/pyccel/pull/420/files/b34803ddda5c4f6d96630d143397ce8ec67fdf2b#diff-d4275df7f6b7ec4eddfc1206284ccf89R203